testing/test_collection.py

```diff
@@ -98,6 +98,14 @@ def pytest_collect_file(path, parent):
         node = testdir.getpathnode(hello)
         assert isinstance(node, pytest.File)
         assert node.name == "hello.xxx"
+
+        # HACK: unset _duplicatepaths, which appears to be necessary for this
+        #       specific test setup?!
+        assert node.session.config.pluginmanager._duplicatepaths == {
+            (type(node), hello)
+        }
+        node.session.config.pluginmanager._duplicatepaths = set()
+
```

Comment:
Will losen that a bit, but still wonder if that's expected/sane.

```diff
diff --git a/testing/test_collection.py b/testing/test_collection.py
index 659d39c8..fbc8cc7b 100644
--- a/testing/test_collection.py
+++ b/testing/test_collection.py
@@ -101,9 +101,7 @@ def pytest_collect_file(path, parent):
 
         # HACK: unset _duplicatepaths, which appears to be necessary for this
         #       specific test setup?!
-        assert node.session.config.pluginmanager._duplicatepaths == {
-            (type(node), hello)
-        }
+        assert len(node.session.config.pluginmanager._duplicatepaths) == 1
```

It is likely due to this specific test setup, right?  (also causes https://github.com/pytest-dev/pytest/pull/4331)